src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardSamUserDirectives.scala

```diff
@@ -49,7 +49,10 @@ trait StandardSamUserDirectives extends SamUserDirectives with LazyLogging with
   def asAdminServiceUser: Directive0 = requireOidcHeaders.flatMap { oidcHeaders =>
     Directives
       .mapInnerRoute { r =>
-        if (!adminConfig.serviceAccountAdmins.contains(oidcHeaders.email)) {
+        if (
+          !adminConfig.serviceAccountAdmins.contains(oidcHeaders.email) &&
+          !adminConfig.serviceAccountAdmins.contains(WorkbenchEmail(s"${oidcHeaders.externalId.map(_.value).toOption.orNull}@uami.terra.bio"))
```

Comment:
It bothers me that if the external id is absent, this ends up checking `null@uami.terra.bio`. I know that that is not real but it also should never be treated as valid. We also should only check for this kind of email if `oidcHeaders.email` is empty.
I suggest using a function like
```
  private def isAdminUami(oidcHeaders: OIDCHeaders) = {
    val maybeUamiEmail = oidcHeaders.externalId.map(b2cId => WorkbenchEmail(s"${b2cId.value}@uami.terra.bio")).toOption
    oidcHeaders.email.value.isEmpty && maybeUamiEmail.isDefined && adminConfig.serviceAccountAdmins.contains(maybeUamiEmail.get)
  }
```

Reply:
Added suggested helper function in [ea35970](https://github.com/broadinstitute/sam/pull/1466/commits/ea359701ed0994c4ab517a75dafe378ab117b05c)